tests/test_index.py

```diff
@@ -222,3 +222,38 @@ def test_search_numpy_filter_str(
                     for doc_id in doc_ids_query_k
                 ]
             )
+
+
+def test_search_numpy_membership_filter(
+    pqlite_with_heterogeneous_tags, heterogenenous_da
+):
+
+    X = np.random.random((Nq, D)).astype(np.float32)
+    query_np = np.array([X[i] for i in range(Nq)])
+    da = heterogenenous_da
+
+    dists, doc_ids = pqlite_with_heterogeneous_tags.search_numpy(
+        query_np,
+        filter={'category': {'$in': ['comics', 'audiobook']}},
+        include_metadata=True,
+    )
+    for doc_ids_query_k in doc_ids:
```

Comment:
assert len of doc_ids otherwise if len doc_ids is 0 the test will give a false positive

Reply:
true, fixed